src/getPage.tsx

```diff
@@ -79,7 +82,11 @@ export default async function getPage({
       <RouterProvider pageObject={pageObject} options={options}>
         <NavigationProvider
           makePage={async (route) => {
-            const { pageElement } = await makePage(route);
+            const { pageElement } = await makePage(route, (req) => {
+              req.headers.cookie = document.cookie;
```

Comment:
Just one further consideration.
At the 1st render we assign  `document.cookie` the value found in `req.headers.cookie`.

From 2 render on, we're preserving cookies information by assigning `req.headers.cookie` `document.cookie` value ( in `makePage`).

So, if I get it right, we have `req.headers.cookie` -> `document.cookie`  -> `req.headers.cookie`.

I mentioned it a few commits ago, but it still seems a valid point to me. Would it make sense to make this flow unidirectional and make `document.cookie` the only source of true of cookie info?

This would involve mocking cookies in `document.cookie` instead of `req.headers.cookie`.

From my point of view it could make sense, since the cookies we're mocking into `req.headers.cookie` would be, in a real application, stored in `document.cookie` and later appended to a `fetch` call.

...or cookies can be handled differently in Next.js?